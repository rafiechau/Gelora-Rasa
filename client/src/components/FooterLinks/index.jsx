import PropTypes from 'prop-types';
import classes from './style.module.scss';

const FooterLinks = ({ title, items }) => (
  <div className={classes.footerLinks} data-testid="footer-links">
    <h4 className={classes.footerLinksTitle} data-testid="footer-links-title">
      {title}
    </h4>
    {items.map((item, index) => (
      <div key={index} className={classes.footerLinkItem} data-testid={`footer-link-item-${index}`}>
        {item}
      </div>
    ))}
  </div>
);

FooterLinks.propTypes = {
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  items: PropTypes.arrayOf(PropTypes.node).isRequired,
};

export default FooterLinks;
