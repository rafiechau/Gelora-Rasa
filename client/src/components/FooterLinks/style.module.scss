.footerLinks {
    margin: 1rem;
    text-align: center;

    .footerLinksTitle {
        font-size: 1.2rem;
        margin-bottom: 0.5rem;
        color: #a8d8ff;
    }

    .footerLinkItem {
        font-size: 0.9rem;
        margin: 0.25rem 0;
        cursor: pointer;
        transition: color 0.3s ease;

        &:hover {
            color: #ffffff;
        }
    }
}