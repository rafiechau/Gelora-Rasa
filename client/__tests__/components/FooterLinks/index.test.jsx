import React from 'react';
import { render, screen } from '@testing-library/react';
import FooterLinks from '@components/FooterLinks';

describe('FooterLinks', () => {
  const mockTitle = 'Test Title';
  const mockItems = ['Item 1', 'Item 2', 'Item 3'];

  beforeEach(() => {
    render(<FooterLinks title={mockTitle} items={mockItems} />);
  });

  test('should render FooterLinks component', () => {
    expect(screen.getByTestId('footer-links')).toBeInTheDocument();
  });

  test('should render correct title', () => {
    expect(screen.getByTestId('footer-links-title')).toHaveTextContent(mockTitle);
  });

  test('should render correct number of items', () => {
    const items = screen.getAllByTestId((_, element) =>
      element.getAttribute('data-testid').startsWith('footer-link-item')
    );
    expect(items).toHaveLength(mockItems.length);
  });

  test('should match snapshot', () => {
    const footerLinks = screen.getByTestId('footer-links');
    expect(footerLinks).toMatchSnapshot();
  });
});
