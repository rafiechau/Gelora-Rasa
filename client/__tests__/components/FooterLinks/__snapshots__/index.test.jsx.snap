// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLinks should match snapshot 1`] = `
<div
  class="footerLinks"
  data-testid="footer-links"
>
  <h4
    class="footerLinksTitle"
    data-testid="footer-links-title"
  >
    Test Title
  </h4>
  <div
    class="footerLinkItem"
    data-testid="footer-link-item-0"
  >
    Item 1
  </div>
  <div
    class="footerLinkItem"
    data-testid="footer-link-item-1"
  >
    Item 2
  </div>
  <div
    class="footerLinkItem"
    data-testid="footer-link-item-2"
  >
    Item 3
  </div>
</div>
`;
